app/services/file_cleanup.py

```
@@ -65,34 +66,65 @@ def check_path_exists(p: Path) -> bool:
+                            db.query(ColdStorageLocation)
+                            .filter(ColdStorageLocation.id == file_record.cold_storage_location_id)
+                            .first()
+                        )
+                        if cold_location:
+                            try:
+                                backend = get_backend(cold_location)
+                                if backend.backend_name() != "local":
+                                    cold_exists = backend.exists(
+                                        file_record.cold_storage_path, cold_location
+                                    )
+                            except Exception as backend_error:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)